i18n: modernize and clean up translations for multiple locales

vivekyadav-3 opened the pull request:
- Cleaned up Hindi (hi) translation file (removed duplicates, fixed indentation)
- Updated Preferences and Nav sections for hi, es-419, zh-CN, it, ja, de, and fr-CA
- Aligned Help menu keys across locales with en-US structure

Fixes #3769
Changes:

I have verified that this pull request:

* [ ] has no linting errors (`npm run lint`)
* [ ] has no test errors (`npm run test`)
* [ ] has no typecheck errors (`npm run typecheck`)
* [ ] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [ ] is descriptively named and links to an issue number, i.e. `Fixes #123`
* [ ] meets the standards outlined in the [accessibility guidelines](https://github.com/processing/p5.js-web-editor/blob/develop/contributor_docs/accessibility.md)

vivekyadav1207vy-sudo committed:
i18n: modernize and clean up translations for multiple locales

- Cleaned up Hindi (hi) translation file (removed duplicates, fixed indentation)
- Updated Preferences and Nav sections for hi, es-419, zh-CN, it, ja, de, and fr-CA
- Aligned Help menu keys across locales with en-US structure

vivekyadav1207vy-sudo committed:
i18n: add missing p5.js 2.0 release keys to Italian translation

vivekyadav1207vy-sudo committed:
Migrate Modal, Toast, and Admonition to TypeScript

vivekyadav1207vy-sudo committed:
Address Copilot feedback: Complete missing translation keys in fr-CA, de, ja, and hi

vivekyadav1207vy-sudo committed:
Migrate TextArea and FloatingActionButton to TypeScript with tests

vivekyadav1207vy-sudo committed:
test: add unit tests for Admonition component

yugalkaushik commented:
Hi @vivekyadav-3 Thank you for the contributions but for translation related contributions we request you only contribute to your native languages. Check out the guidelines here: https://github.com/processing/p5.js-web-editor/blob/develop/contributor_docs/translations.md , Along with it please run the test cases.

vivekyadav1207vy-sudo committed:
refactor: remove non-native translations and keep only Hindi as requested

vivekyadav-3 commented:
"Hi @yugalkaushik, thank you for the feedback!
i have removed the non-native translations and kept only the Hindi ones.I
have also verified the project structure locally.
Ready for another review.Thanks for the guidance."


On Sun, Dec 28, 2025 at 6:07 PM Yugal Kaushik ***@***.***>
wrote:

> *yugalkaushik* left a comment (processing/p5.js-web-editor#3769)
> <https://github.com/processing/p5.js-web-editor/pull/3769#issuecomment-3694715792>
>
> Hi @vivekyadav-3 <https://github.com/vivekyadav-3> Thank you for the
> contributions but for translation related contributions we request you only
> contribute to your native languages. Check out the guidelines here:
> https://github.com/processing/p5.js-web-editor/blob/develop/contributor_docs/translations.md
> , Along with it please run the test cases.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/processing/p5.js-web-editor/pull/3769#issuecomment-3694715792>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BSGQEXFB6OVFG4UJN2AFZQL4D7FKNAVCNFSM6AAAAACQCEOKDKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMOJUG4YTKNZZGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 


* <http://www.kiit.ac.in>*

*
Go Green: Kindly don't print this unless so 
required.*

*
*

*Established U/S 3 of UGC Act and Accredited by NBA of 
AICTE and NAAC of UGC*



Visit us @ *http://www.kiit.ac.in 
<http://www.kiit.ac.in/>*

Follow us @* **|***Twitter 
<https://www.twitter.com/KIITUniversity/>***|***Facebook** 
<https://www.facebook.com/KIITUniversity>*|***Instagram 
<https://www.instagram.com/KIITUniversity/>***|*


 
  
  

The information 
contained in this electronic message and any
  attachments to this message 
are intended for the exclusive use of the
  addressee(s) and may contain 
proprietary, confidential or privileged
  information. If you are not the 
intended recipient, you should not
  disseminate, distribute or copy this 
e-mail. Please notify the sender
  immediately and delete all copies of 
this message and any attachments.
  

 VIRUS WARNING: Computer viruses can 
be
  transmitted via email. The recipient should check this email and any
  
attachments for the presence of viruses. The institute accepts no liability
  for any damage caused by any virus transmitted by this email.

Jatin24062005 commented:
@yugalkaushik  is right ! @vivekyadav-3   this is definitely useful 
However, since this PR touches many language files, reviewing and verifying accuracy becomes quite difficult without native or local-language verification. For translations, we strongly prefer that changes can be validated by someone who actually speaks the language, not just structurally aligned with en-US. 
please limit this PR to at most 1–2 languages (ideally ones you or another contributor can confidently verify),

vivekyadav1207vy-sudo committed:
i18n: restore Hindi locale from origin and add missing keys

vivekyadav-3 commented:
Hi @Jatin24062005, thank you for the feedback!

I have updated the PR to limit the changes to English and Hindi, as I am a native Hindi speaker and can confidently verify them. I've reverted the changes to all other language files to keep the review process manageable and accurate.

I also fixed a syntax error I found in the Hindi locale and ensured all keys are now synced with the English version. Ready for another look!
